src/sentry/notifications/utils/participants.py

```diff
@@ -85,51 +86,89 @@ def get_participants_for_group(
     return participants_by_provider
 
 
-def get_reason(
-    user: User, value: NotificationSettingOptionValues, user_ids: set[int]
-) -> int | None:
-    # Members who opt into all deploy emails.
-    if value == NotificationSettingOptionValues.ALWAYS:
-        return GroupSubscriptionReason.deploy_setting
-
-    # Members which have been seen in the commit log.
-    elif value == NotificationSettingOptionValues.COMMITTED_ONLY and user.id in user_ids:
-        return GroupSubscriptionReason.committed
-    return None
-
-
 def get_participants_for_release(
-    projects: Iterable[Project], organization: Organization, user_ids: set[int]
+    projects: Iterable[Project],
+    organization: Organization,
+    user_ids: set[int],
+    notification_setting_type: NotificationSettingTypes = NotificationSettingTypes.DEPLOY,
 ) -> Mapping[ExternalProviders, Mapping[Team | User, int]]:
     # Collect all users with verified emails on a team in the related projects.
     users = set(User.objects.get_team_members_with_verified_email_for_projects(projects))
 
-    # Get all the involved users' settings for deploy-emails (including
-    # users' organization-independent settings.)
-    notification_settings = NotificationSetting.objects.get_for_recipient_by_parent(
-        NotificationSettingTypes.DEPLOY,
-        recipients=users,
-        parent=organization,
-    )
-    notification_settings_by_recipient = transform_to_notification_settings_by_recipient(
-        notification_settings, users
+    parent_specific_scope_type = get_scope_type(notification_setting_type)
+
+    def parent_notification_settings_by_recipient(
+        note_setting_type: NotificationSettingTypes,
```

Comment:
nit: would it be better to call this setting_type since it's already in the context of notifications?